### DISCUSSION TITLE
How to insert an optional value

### DISCUSSION THREAD

**Discussion by Overdash** [SeaQL/sea-query#843] [Q&A]
I have this query - `last_name` and `phone_number` are `Option<String>` 

```rust
let (sql, values) = Query::insert()
            .into_table(AccountDef::Table)
            .columns([
                AccountDef::AccountId,
                AccountDef::AuthId,
                AccountDef::Email,
                AccountDef::FirstName,
                AccountDef::LastName,
                AccountDef::PhoneNumber,
            ])
            .values([
                self.account_id.into(),
                self.email,
                self.first_name,
                self.last_name.into(),
                self.phone_number.into(),
            ])?
            .returning_all()
            .build_sqlx(PostgresQueryBuilder)
```

I get this error
```
the trait `From<std::option::Option<std::string::String>>` is not implemented for `std::string::String`, which is required by `std::option::Option<std::string::String>: Into<_>`
```
Is there a way to actually do this, so that optional values can be used?
The example seem quite lacking in this front. And there's no clear discussion on this since this [unresolved thread](https://github.com/SeaQL/sea-query/discussions/734).

**Comment by Expurple** [comment]
SeaQuery should support this. This looks like a type inference issue in your code. If `self.email` is a `String`, then the array of values is inferred to be a `[String; 5]`. That's why `.into()` attempts to convert the other array elements to `String`. But [`values()`](https://docs.rs/sea-query/0.32.0/sea_query/query/struct.InsertStatement.html#method.values) accepts an `IntoIterator<Item = SimpleExpr>`, so you actually need a `[SimpleExpr; 5]`. Try giving the compiler a cue that you're mapping `T -> Value -> SimpleExpr`:
```rust
            .values([
                self.account_id.into(),
                self.email.into(),
                self.first_name.into(),
                self.last_name.into(),
                self.phone_number.into(),
            ].map(SimpleExpr::Value))?
```
But I think, SeaQuery's API could be improved. It could accept something like `IntoIterator<Item: Into<SimpleExpr>>`. In that case, your code would only need to build the `Value`s, and then the method would convert them into `SimpleExpr`s under the hood.
